protocols/v2/roles-logic-sv2/src/job_creator.rs

```diff
@@ -683,19 +463,32 @@ pub mod tests {
             235, 216, 54, 151, 78, 140, 249, 1, 32, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         ];
-        let coinbase: Transaction = consensus::deserialize(encoded).unwrap();
-        let stripped = StrippedCoinbaseTx::from_coinbase(coinbase.clone(), 32).unwrap();
-        let prefix = stripped.into_coinbase_tx_prefix().unwrap().to_vec();
-        let suffix = stripped.into_coinbase_tx_suffix().unwrap().to_vec();
+        let tx: Transaction = consensus::deserialize(encoded).unwrap();
+
+        // Calculate script_sig_prefix_len by examining the actual script_sig length
+        // and subtracting the extranonce length (32 in this case)
+        let extranonce_len = 32;
+        let script_sig_len = tx.input[0].script_sig.len();
+        let script_sig_prefix_len = script_sig_len - extranonce_len;
```

Comment:
would it make sense to save the right side (`extranonce_len` here) instead of `script_sig_prefix_len`? it feels like we are doing this calculation but are not using it until we call other function. maybe the responsibility of doing this should be on the other function.